### PR TITLE
Added font weight palette and update H1 font weight

### PR DESCRIPTION
**Pull request by origami-z**
H1 should be the only one changed

Display 4 / amplitude font weight switch
- step 1: this PR commit 1, add font weight palette for theme 1
- step 2: this PR commit 2, add font weight switch at palette level for theme 2
- step 3: add display 4 mapping #3495

Part of #3368

### CODE REVIEW

File: .changeset/lucky-apricots-try.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+| Default | Bold       | Semi bold |
+| Small   | Medium     | Regular   |
+| Strong  | Extra bold | Bold      |
+
+Added font weight in palette layer for display, heading, body and notation.
+
+```diff
++ --salt-palette-text-display-fontWeight: var(--salt-typography-fontWeight-semiBold);
++ --salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);
++ --salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
```
Comment (joshwooding) [inline .changeset/lucky-apricots-try.md:18]:
```suggestion
+ --salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
+ --salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);
```
The order was confusing me :)

File: packages/theme/css/palette/text-next.css [hunk 2]
```diff
@@ -1,7 +1,39 @@
 .salt-theme-next.salt-theme[data-heading-font="Open Sans"] {
   --salt-palette-text-fontFamily-heading: var(--salt-typography-fontFamily-openSans);
+
+  --salt-palette-text-display-fontWeight: var(--salt-typography-fontWeight-semiBold);
+  --salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);
+  --salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
```
Comment (joshwooding) [inline packages/theme/css/palette/text-next.css:6]:
```suggestion
  --salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
  --salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);
```

**Review by joshwooding — APPROVED** [review]

File: packages/core/stories/salt-provider/salt-provider-next.mdx [hunk 3]
```diff
@@ -169,6 +169,10 @@ between Open Sans and Amplitude.
 <UNSTABLE_SaltProviderNext headingFont="Amplitude">
 ```
 
+When Amplitude is used, default heading font weight is set to meidum (instead
```
Comment (pseys) [inline packages/core/stories/salt-provider/salt-provider-next.mdx:172]: Spelling error. 'meidum'

Comment (origami-z) [reply packages/core/stories/salt-provider/salt-provider-next.mdx:172]:
```suggestion
When Amplitude is used, default heading font weight is set to medium (instead
```

File: packages/theme/css/palette/text.css [hunk 4]
```diff
@@ -2,4 +2,20 @@
   --salt-palette-text-fontFamily: var(--salt-typography-fontFamily-openSans);
   --salt-palette-text-fontFamily-heading: var(--salt-typography-fontFamily-openSans);
   --salt-palette-text-fontFamily-code: var(--salt-typography-fontFamily-ptMono);
+
+  --salt-palette-text-display-fontWeight: var(--salt-typography-fontWeight-semiBold);
+  --salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);
+  --salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
```
Comment (pseys) [inline packages/theme/css/palette/text.css:8]: --salt-palette-text-display-fontWeight: var(--salt-typography-fontWeight-semiBold);
--salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
--salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);

**Review by pseys — COMMENTED** [review]
Left a couple of comments

File: packages/theme/css/palette/text.css (hunk 4 above)
Comment (origami-z) [inline packages/theme/css/palette/text.css:8]:
```suggestion
  --salt-palette-text-display-fontWeight: var(--salt-typography-fontWeight-semiBold);
  --salt-palette-text-display-fontWeight-small: var(--salt-typography-fontWeight-regular);
  --salt-palette-text-display-fontWeight-strong: var(--salt-typography-fontWeight-bold);
```

**Review by pseys — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**